src/shared/Mongo/MongoConnector.ts

```diff
@@ -0,0 +1,66 @@
+import { MongoClient, Db } from 'mongodb';
+import { reportError } from '../../drivers/SentryConnector';
+import { ServiceError, ServiceErrorReason } from '../errors';
+
+export class MongoConnector {
+  private static mongoClient: MongoClient;
+
+  /**
+   * Opens a new connection to the database that can be shared via
+   * the getConnection function.
+   * @param dburi the URI of the MongoDB instance to connect to.
+   */
+  static async open(dburi: string): Promise<void> {
+    try {
+      if (!MongoConnector.mongoClient) {
+        const driver = new MongoConnector();
+        await driver.connect(dburi);
+      } else {
+        return Promise.reject(
+          new Error('There can be only one MongoClient'),
+        );
+      }
+    } catch (error) {
+      reportError(error);
+      return Promise.reject(
+        new ServiceError(
+          ServiceErrorReason.INTERNAL,
+        ),
+      );
+    }
+  }
+
+  /**
+   * Retrieves the client with an open connection to the database.
+   */
+  static client(): MongoClient {
+    return this.mongoClient;
+  }
+
+  /**
+   * Connect to the database. Must be called before any other functions.
+   * @param {string} dbIP the host and port on which mongodb is running
+   */
+  private async connect(dbURI: string, retryAttempt?: number): Promise<void> {
+    try {
+      MongoConnector.mongoClient = await MongoClient.connect(dbURI);
```

Comment:
We should use MongoClient's baked in retry mechanism instead of the home baked one.
https://mongodb.github.io/node-mongodb-native/api-generated/mongoclient.html

```suggestion
      MongoConnector.mongoClient = await new MongoClient(dbURI, {reconnectTries: 1).connect();
```